Training24/Test.c

```diff
@@ -90,97 +96,116 @@ void PrintTableHeader (enum HeaderType type) {
+            if (!overflow) {
+               strcpy (outputStr, "Integer Overflow");
+               const char* actualOutput = "Failed to process";
+               PrintIntegerTableRow (number, outputStr, expectedOutputs[i], actualOutput, testCases[i]);
+            }
+            else {
+               sprintf (outputStr, "%d", number);
+               const char* actualOutput = IsIntPalindrome (number) ? "Palindrome" : "Not a Palindrome";
+               PrintIntegerTableRow (number, outputStr, expectedOutputs[i], actualOutput, testCases[i]);
+            }
          }
       }
    }
 }
 
-void TestUserPhrase () {
+void GetUserPhrase () {
    char userPhrase[500];
    while (1) {
-      printf ("\nEnter a phrase to test (max 499 characters, or type 'r' to return menu): ");
+      printf ("\nEnter a phrase to test (max 499 characters, or type 'r' or 'R' to return menu): ");
       if (fgets (userPhrase, sizeof (userPhrase), stdin) == NULL) {
          printf (YELLOW "Error: Input not received. Exiting function." RESET "\n");
          return;
       }
       userPhrase[strcspn (userPhrase, "\n")] = 0;
-      if (strcmp (userPhrase, "r") == 0 || strcmp (userPhrase, "R") == 0) return;   // Return to menu command
+      if (!strcmp (userPhrase, "r") || !strcmp (userPhrase, "R")) return;
       if (strlen (userPhrase) >= sizeof (userPhrase) - 1) {
          printf (YELLOW "Error: Input exceeds maximum length." RESET "\n");
          while (getchar () != '\n' && getchar () != EOF);
          continue;
       }
-      else printf ("Result: %s\n", CheckPhrasePalindrome (userPhrase) ? GREEN "It's a Palindrome" RESET : RED "Not a Palindrome" RESET);
+      if (strlen (userPhrase) == 0) {
+         printf ("Result: %s\n", GREEN "It's a Palindrome" RESET);
+         continue;
+      }
+      printf ("Result: %s\n", IsPhrasePalindrome (userPhrase)
+         ? GREEN "It's a Palindrome" RESET
+         : RED "Not a Palindrome" RESET);
    }
 }
 
-void TestUserInteger () {
+void GetUserInteger () {
    char userInput[256];
    while (1) {
-      printf ("\nEnter a positive integer to test or type 'r' to return: ");
+      printf ("\nEnter a positive integer to test or type 'r' or 'R' to return menu: ");
       if (!fgets (userInput, sizeof (userInput), stdin) || userInput[0] == '\n' || userInput == NULL) {
          printf ("Result: %s\n", YELLOW "Invalid input (Empty string)" RESET);
          continue;
       }
-      if (strcmp (userInput, "r\n") == 0 || strcmp (userInput, "R\n") == 0) return;   // Return to menu command
+      if (!strcmp (userInput, "r\n") || !strcmp (userInput, "R\n") ) return;   // Return to menu command
       char* endPtr;
-      long inputNumber = strtol (userInput, &endPtr, 10);
-      if (endPtr == userInput || *endPtr != '\n' || inputNumber < 0) {   // Check if input is valid
+      int inputNumber = strtol (userInput, &endPtr, 10);
+      if (endPtr == userInput || *endPtr != '\n' || inputNumber < 0) {
```

Comment:
Negative number check must be done by your function, not outside of it. A negative number is not invalid. It is simply not a palindrome.